### PR TITLE
Optimize API Changes

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Issue:
1) It is hard to follow what is going on as we are throwing all optimize interventions into one big blob.
2) Our optimize intervention type (find new start time, param value ect) was being incorrectly assigned when multiple types were given all but the first were ignored.

# Description
- Re-organize how we are sending in the interventions to be optimized.
    - Rather than sending one big messy blob send a list of `InterventionObjective`
- Utilize `intervention_func_combinator` to allow us to intervene on two interventions with different types.
    - Previously if we sent the screenshot below both of the interventions were incorrectly using `new start time`
- Provide `objfun` our list of interventions to optimize over as in follow up PR we will be fixing and cleaning the logic written here but I do not want to flood this PR with more changes as it may become very hard to follow.
The follow up PR will solve the issue that asking to optimize for `start time AND value` fails 

# Related Required:
 https://github.com/DARPA-ASKEM/terarium/pull/5360

# Pictures
image

### CODE REVIEW

File: service/models/operations/optimize.py [hunk 1]
```diff
@@ -59,23 +61,26 @@ def gen_risk_bound(self):
...
     total_sum = 0
-
+    # TODO: Will be cleaning this up in the next PR. I want minimal changes per PR for readability.
+    relative_importance = [i.relative_importance for i in optimize_interventions]
+    initial_guess = [i.initial_guess for i in optimize_interventions]
+    objective_function_option = [
+        i.objective_function_option for i in optimize_interventions
+    ]
     # Calculate the sum of all weights, fallback to 1 if the sum is 0
     sum_of_all_weights = np.sum(relative_importance) or 1
```
Comment (liunelson) [inline service/models/operations/optimize.py:85]: Use `1.0` instead of `1` to avoid integer division, just in case.

Comment (Tom-Szendrey) [reply service/models/operations/optimize.py:85]: I wanted to keep this PR smaller for readability. The scope of this is to clean, and correctly use multiple intervention types
I will update the objective function in the follow up PR 
I will update this in my follow up branch good call

**Review by liunelson — APPROVED** [review]
Looks great - thanks for tackling this tricky feature. I see that you've covered the last-day vs. global max QoI, min vs. max objective function and the `func_lengths` details correctly (at least as far as I know).
